src/net_processing.cpp

```diff
@@ -2629,8 +2631,10 @@ void ProcessMessage(
                 State(pfrom.GetId())->fProvidesHeaderAndIDs = true;
                 State(pfrom.GetId())->fWantsCmpctWitness = nCMPCTBLOCKVersion == 2;
             }
-            if (State(pfrom.GetId())->fWantsCmpctWitness == (nCMPCTBLOCKVersion == 2)) // ignore later version announces
+            if (State(pfrom.GetId())->fWantsCmpctWitness == (nCMPCTBLOCKVersion == 2)) { // ignore later version announces
                 State(pfrom.GetId())->fPreferHeaderAndIDs = fAnnounceUsingCMPCTBLOCK;
+                pfrom.m_highbandwidth_from = fAnnounceUsingCMPCTBLOCK;
```

Comment:
f9cc66bc An explanatory comment here may be nice; I found this the least clear line of this PR changeset.

Reply:
Done.